### PR TITLE
Package directory structure documantation

### PR DESCRIPTION
**Pull request by Itay4**
fixes https://github.com/demisto/etc/issues/19040

@kirbles19 please review the MD files

### CODE REVIEW

File: docs/package_directory_structure/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+This is required for running [linting](https://github.com/demisto/content/blob/master/docs/linting/README.md) and [unit testing](https://github.com/demisto/content/blob/master/docs/tests/unit-testing/README.md) on the Python code.
+
+The package is converted into a YAML file, which include all the package components, in Demisto's CI build, using the `package_creator.py` script, and then populated in the Content release.
+
+---
+
+A package should be a sub-directory of either Integrations or Scripts directory, and contain the following files:
+
+ - 2 Python files - the code and the unit test code.
```
Comment (ronykoz) [inline docs/package_directory_structure/README.md]: You mingle between file types and file names, choose one

File: docs/package_directory_structure/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ - YAML file - configuration file.
+ - PNG file - integration logo (for integrations only).
+ - MD file - detailed instructions (for integrations only).
+ - CHANGELOG - a markdown file which include the script/integration release notes.
+ - Pipfile (can be copied from Tests/scripts/dev_envs/default_python3)
```
Comment (ronykoz) [inline docs/package_directory_structure/README.md]: Add an explanation about python2 being deprecated, hence the python3

File: docs/package_directory_structure/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+ - `PaloAltoNetworksCortex_test.py` - the integration unit tests.
+ - `PaloAltoNetworksCortex_description.md` - the integration detailed instructions.
+ - `Pipfile`
+ - `Pipfile.lock`
+ 
+---
+
+You can extract a package from a YAML file by using the following:
+ - [Demisto IntelliJ Plugin](https://plugins.jetbrains.com/plugin/12093-demisto-add-on-for-pycharm)
+ - `package_extractor` script
```
Comment (ronykoz) [inline docs/package_directory_structure/README.md]:
```suggestion
 - `package_extractor.py` script from the Content repository 
```

File: docs/package_directory_structure/README.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# Package directory structure
+Content entities in Demisto are presented by YAML files.
```
Comment (yaakovi) [inline docs/package_directory_structure/README.md]: content code entities.
some of our entities are in json.

File: docs/package_directory_structure/README.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+Automation scripts and integrations [YAML files](https://github.com/demisto/content/blob/master/docs/yaml-file-integration/README.MD) include Javascript/Python code.
+
+Python scripts and integrations are stored in directories as packages.
+
+This is required for running [linting](https://github.com/demisto/content/blob/master/docs/linting/README.md) and [unit testing](https://github.com/demisto/content/blob/master/docs/tests/unit-testing/README.md) on the Python code.
+
+The package is converted into a YAML file, which include all the package components, in Demisto's CI build, using the `package_creator.py` script, and then populated in the Content release.
```
Comment (yaakovi) [inline docs/package_directory_structure/README.md]: one can also use the plug-in with the `Export to Demisto` button

File: docs/package_directory_structure/README.md (hunk 4 above)
Comment (ronykoz) [inline docs/package_directory_structure/README.md]: Something feels off in this paragraph, go over it with Alex

File: docs/package_directory_structure/README.md (hunk 1 above)
Comment (yaakovi) [inline docs/package_directory_structure/README.md]: specify the file name standard: `_image.png`, `_description.md`, `_test.py`

File: docs/package_directory_structure/README.md (hunk 5 above)
Comment (Itay4) [reply docs/package_directory_structure/README.md]: yea i wrote that in the bottom
the sentence is here just to clarify how eventually the customer gets a yml out of the package

File: docs/package_directory_structure/README.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+---
+
+A package should be a sub-directory of either Integrations or Scripts directory, and contain the following files:
+
+ - `<INTEGRATION-NAME>.py` - Integration / automation script Python code.
```
Comment (anara123) [inline docs/package_directory_structure/README.md]: can we provide something similar to the following docs
image
https://code.visualstudio.com/api/get-started/extension-anatomy

File: docs/package_directory_structure/README.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+ - `PaloAltoNetworksCortex_test.py` - the integration unit tests.
+ - `PaloAltoNetworksCortex_description.md` - the integration detailed instructions.
+ - `Pipfile`
+ - `Pipfile.lock`
+ 
+---
+
+You can extract a package from a YAML file by using the following:
+ - [Demisto IntelliJ Plugin](https://plugins.jetbrains.com/plugin/12093-demisto-add-on-for-pycharm)
+ -  `package_extractor.py` script from the Content repository.
```
Comment (anara123) [inline docs/package_directory_structure/README.md]: provide full functional example for how to use the script
`mkdir Integrations/Jira && python3 package_extractor.py -i /some/folder/integration-Jira.yml -o Integrations/Jira -t integration -m`

Comment (yaakovi) [reply docs/package_directory_structure/README.md]: no need to do `mkdir`, the script handles that already

**Pull request merged**